sympy/physics/units/util.py

```diff
@@ -120,21 +123,43 @@ def get_total_scale_factor(expr):
     return expr_scale_factor * Mul.fromiter((1/get_total_scale_factor(u) * u) ** p for u, p in zip(target_units, depmat))
 
 
-def quantity_simplify(expr):
+def _build_preferred_dimension_to_unit(unit_system="SI") -> tDict[Symbol, Quantity]:
+    from sympy.physics.units import  velocity, length, time
+
+    unit_system = UnitSystem.get_unit_system(unit_system)
+
+    preferred_dimension_to_unit = {}
+
+    for quantity in unit_system.get_units_non_prefixed():
```

Comment:
You bring up a good point. The purpose of this is to build the lookup table using base units, because otherwise it'll end up picking units like `picometer` or `picojoule`. As a result, `gram` will be picked over `kilogram`, because `kilogram` is prefixed.

I think it would be pretty feasible to grab the degree of the resulting number, and select the closest prefix. However, I feel like it could lead to unexpected behavior. It's extra complexity to debug later, and it shouldn't go in this function. In the interest of trying to keep this PR small, I was going to follow up this PR with another to add a new function or optional arg to this function that does this.

Reply:
Maybe the preferred units for a given unit system should just be hard-coded into a dictionary in the end?

Reply:
But then it won't be unit system agnostic.